apps/account-api/src/services/siwfV2.service.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { ConfigModule } from '@nestjs/config';
+import { decodeSignedRequest } from '@projectlibertylabs/siwfv2';
+import { SiwfV2Service } from './siwfV2.service';
+import apiConfig from '#account-api/api.config';
+import blockchainConfig from '#account-lib/blockchain/blockchain.config';
+import { WalletV2RedirectResponseDto } from '#types/dtos/account/wallet.v2.redirect.response.dto';
+import { BlockchainService } from '#account-lib/blockchain/blockchain.service';
+
+const exampleCallback = 'https://www.example.com/callback';
+const examplePermissions = [1, 3, 5, 7, 9];
+const exampleCredentials = [
+  'VerifiedGraphKeyCredential',
+  'VerifiedEmailAddressCredential',
+  'VerifiedPhoneNumberCredential',
+];
+
+describe('SiwfV2Service', () => {
+  let siwfV2Service: SiwfV2Service;
+
+  const mockBlockchainService = {
+    getNetworkType: jest.fn(),
+  };
+
+  const mockBlockchainServiceProvider = {
+    provide: BlockchainService,
+    useValue: mockBlockchainService,
+  };
+
+  beforeAll(() => {
+    process.env.SIWF_V2_URL = 'https://www.example.com/siwf';
+    process.env.FREQUENCY_URL = 'http://127.0.0.1:9944';
+    process.env.FREQUENCY_HTTP_URL = 'http://127.0.0.1:9944';
+    process.env.PROVIDER_ACCOUNT_SEED_PHRASE =
+      'offer debate skin describe light badge fish turtle actual inject struggle border';
+    process.env.PROVIDER_ID = '1';
+    process.env.WEBHOOK_BASE_URL = 'http://127.0.0.1';
+    process.env.CAPACITY_LIMIT = '{"type":"amount","value":"80"}';
+    process.env.GRAPH_ENVIRONMENT_TYPE = 'TestnetPaseo';
+  });
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        ConfigModule.forRoot({
+          load: [apiConfig, blockchainConfig],
+        }),
+      ],
+      providers: [SiwfV2Service, mockBlockchainServiceProvider],
+    }).compile();
+
+    siwfV2Service = module.get<SiwfV2Service>(SiwfV2Service);
+  });
+
+  it('should be defined', () => {
+    expect(siwfV2Service).toBeDefined();
+  });
+
+  describe('getRedirectUrl', () => {
+    let result: WalletV2RedirectResponseDto;
+
+    beforeAll(async () => {
+      result = await siwfV2Service.getRedirectUrl(exampleCallback, examplePermissions, exampleCredentials);
+    });
```

Comment:
Since this function has the potential to throw, it would be good to test that it doesn't before proceeding.
```suggestion
    it('Should not throw', async () => {
      expect.assertions(1);
      result = await siwfV2Service.getRedirectUrl(exampleCallback, examplePermissions, exampleCredentials);
      expect(result).toBeDefined();
    });
```

Jest runs all tests in the same `describe` block serially (and awaits any async tests).

Reply:
A thrown error on a jest test beforeAll still counts as a failure. Do we think that isn't good enough?

```
 FAIL  apps/account-api/src/services/siwfV2.service.spec.ts (5.296 s)
  ● SiwfV2Service › getRedirectUrl › Should return a valid signedRequest

    Failed to get SIWF V2 Redirect URL
```